content/docs/refs-and-the-dom.md

```diff
@@ -11,33 +11,33 @@ redirect_from:
 permalink: docs/refs-and-the-dom.html
 ---
 
-Refs provide a way to access DOM nodes or React elements created in the render method.
+تُزوّدنا المراجع بطريقة للوصول إلى عقد DOM أو عناصر React المُنشأة باستخدام التابع render.
 
-In the typical React dataflow, [props](/docs/components-and-props.html) are the only way that parent components interact with their children. To modify a child, you re-render it with new props. However, there are a few cases where you need to imperatively modify a child outside of the typical dataflow. The child to be modified could be an instance of a React component, or it could be a DOM element. For both of these cases, React provides an escape hatch.
+تكون [الخاصيات](/docs/components-and-props.html) في تدفق بيانات React النموذجي هي الطريقة الوحيدة التي يتواصل بها المكون الأب مع ابنائه. وللتعديل على أبن يتم اعادة تصيير المكون بخاصيات جديدة. على الرغم من ذلك ، في بعض الحالات التي قد تحتاج الى اجراء تعديل على مكون أبن بشكل إجباري خارج نطاق التدفق المعتاد. من الممكن ان يكون المكون الابن المراد تعديله نسخة instance من مكون React ، أو ان يكون عنصر DOM. في كلتا الحالتين توفر لنا React مفر للألتفاف حول هذه الحالات.
```

Comment:
Isn't `تُعتَبرُ` more correct to `تكون` in this sentence?
```
تكون الخاصيات في تدفق بيانات 
```
---
In `وللتعديل على أبن ` did you mean `وللتعديل على الإبن` or `وللتعديل على العنصر الإبن`? (You wrote `أبن` instead of `إبن`)
---
Also in `في كلتا الحالتين توفر لنا React مفر للألتفاف حول هذه الحالات.`, it should be `للإلتفاف`

```suggestion
تكون [الخاصيات](/docs/components-and-props.html) في تدفق بيانات React النموذجي هي الطريقة الوحيدة التي يتواصل بها المكون الأب مع أبنائه. وللتعديل على الإبن يتم اعادة تصيير المكون بخاصيات جديدة. على الرغم من ذلك ، في بعض الحالات التي قد تحتاج الى اجراء تعديل على مكون إبن بشكل إجباري خارج نطاق التدفق المعتاد. من الممكن ان يكون المكون الإبن المُراد تعديله نسخة instance من مُكوّن React ، أو ان يكون عنصر DOM. في كلتا الحالتين توفر لنا React مفر للإلتفاف حول هذه الحالات.
```

Reply:
yes, you're absolutely right, I might have missed those because of how excited I was to get this repo to 70% so that it can be published, I finished that page in like 2 days